clang/include/clang/Basic/Attr.td

```diff
@@ -1082,6 +1082,13 @@ def SYCLDevice : InheritableAttr {
   let Documentation = [SYCLDeviceDocs];
 }
 
+def SYCLAllowRawPointerInKernel : Attr {
```

Comment:
Not sure about the naming. But your attribute is not for kernels, it's for `SYCL_EXTERNAL` marked functions, right?
```suggestion
def SYCLAllowRawPointerInSYCLExternal: Attr {
```